esmvaltool/diag_scripts/hydrology/globwat.py

```diff
@@ -0,0 +1,346 @@
+    """Change negative values to zero."""
+    cube.data = da.where(cube.core_data() < 0, 0, cube.core_data())
+    return cube
+
+
+def get_input_cubes(metadata):
+    """Return a dictionary with all (preprocessed) input files."""
+    provenance = create_provenance_record()
+    all_vars = {}
+    for attributes in metadata:
+        short_name = attributes['short_name']
+        filename = attributes['filename']
+        logger.info("Loading variable %s", short_name)
+        cube = iris.load_cube(filename)
+        all_vars[short_name] = change_data_type(cube)
+        cube.attributes['mip'] = attributes['mip']
+        provenance['ancestors'].append(filename)
+    return all_vars, provenance
+
+
+def load_target(cfg):
+    """Load target grid."""
+    filename = Path(cfg['auxiliary_data_dir']) / cfg['target_file']
+    cube = iris.load_cube(str(filename))
+    for coord in 'longitude', 'latitude':
+        if not cube.coord(coord).has_bounds():
+            cube.coord(coord).guess_bounds()
+    return cube
+
+
+def langbein_pet(tas):
+    """Calculate potential ET using Langbein method.
+
+    The Langbein curve represents an empirical relationship between temperature
+    and potential ET (pet). Where T is the annual average temperature in degree
+    Celcius, pet is in mm per year, and a, b, c are unitless empirical
+    constants.
+    Reference: https://doi.org/10.3133/cir52 page 8, figure 1.
+    An example of using Langbein method can be found at:
+    https://doi.org/10.1080/02626667.2017.1332416 page 1472, equation 7.
+    """
+    tas.convert_units('degC')
+    constant_a = iris.coords.AuxCoord(np.float32(325),
+                                      long_name='first constant', units=None)
+    constant_b = iris.coords.AuxCoord(np.float32(21),
+                                      long_name='second constant', units=None)
+    constant_c = iris.coords.AuxCoord(np.float32(0.9),
+                                      long_name='third constant', units=None)
+
+    # assumption here: tas is constant over time, then the monthly/daily
+    # average value is equal to the annual average.
+    pet = (tas) * constant_b + (tas ** 2) * constant_c + constant_a
+    pet.units = 'kg m-2 year-1'  # equivalent to mm year-1
+    pet.convert_units('kg m-2 s-1')  # convert to a cmor compatible unit
+    pet.var_name = 'evspsblpot'
+    pet.standard_name = 'water_potential_evaporation_flux'
+    pet.long_name = 'Potential Evapotranspiration'
+    return pet
+
+
+def get_cube_time_info(cube):
+    """Return year, month and day from the cube."""
+    coord_time = cube.coord('time')
+    time = coord_time.cell(0).point
+    time_step = time.strftime("%Y%m%d")
+    return time_step
+
+
+def get_cube_data_info(cube):
+    """Return short_name, and mip from the cube."""
+    short_name = cube.var_name
+    mip = cube.attributes['mip']
+    return short_name, mip
+
+
+def _reindex_data(cube, target):
+    """Reindex data to a global coordinates set.
```

Comment:
I'm not sure I understand what this is doing. The incoming data is already on a global coordinates set, right?

Reply:
If users use an area-selection preprocessor, the data should be reindexed to a global extent.

Reply:
I'm sorry, this is still unclear to me. Why would you use area-selection at all if you're going to change it back to a global extent anyway?

Reply:
right, the `_reindex_data()` is removed now.